### PR TITLE
sorter(ticdc): use correct cgroup memory limit in some scenarios

### LINKED ISSUE #8588: cgroup memory limit is not awared in some scenarios
### What did you do?

- Deploy ticdc server with tiup
- Add `MemoryMax=15G` in systemd configuration.

```bash
$ cat /etc/systemd/system/cdc-8301.service
[Unit]
Description=cdc service
After=syslog.target network.target remote-fs.target nss-lookup.target

[Service]
LimitNOFILE=1000000
LimitSTACK=10485760
User=apple
ExecStart=/bin/bash -c '/home/apple/deploy/cdc-upstream-1/tidb-deploy/cdc-8301/scripts/run_cdc.sh'
Restart=always

RestartSec=15s
MemoryMax=15G

[Install]
WantedBy=multi-user.target
```
- Restart ticdc server

### What did you expect to see?

The memory limit of cdc server should be `15G`

Instead it uses host memory as total memory, and sets memory limit to 6.3G in sorter engine

```go
[2023/03/20 12:15:40.097 +08:00] [INFO] [server.go:210] ["sorter engine memory limit"] [memory="6.3 GB"]
```

```bash
$ free -h
               total        used        free      shared  buff/cache   available
Mem:            58Gi       6.4Gi        47Gi        15Mi       5.2Gi        51Gi
Swap:             0B          0B          0B
```

### What did you see instead?

TiCDC server doesn't find the correct memory limit, and uses total memory of host instead.

It is also a known bug in TiDB library: https://github.com/pingcap/tidb/issues/38659

### Versions of the cluster

TiCDC version (execute `cdc version`):

master@https://github.com/pingcap/tiflow/commit/41b40aaff95f8754eea7ec792135debe70b8e4da

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #8588

### What is changed and how it works?

Use a better way to fetch cgroup memory limit.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Have been verified in the following scenarios

- [x] systemd cgroup enabled
- [x] systemd cgroup not enabled
- [x] `docker run` container under `containerd-shim-runc-v2` with memory limit
- [x] `docker run` container under `containerd-shim-runc-v2` without memory limit
- [x] K8s starts pod with memory limit
- [x] K8s starts pod without memory limit 

### Release note 

```release-note
Use correct cgroup memory limit in sorter engine under some scenarios.
```

### CODE REVIEW

**Review by hicqu — APPROVED** [review]

File: pkg/util/memory.go
Comment (CharlesCheung96) [inline pkg/util/memory.go:1]: Maybe `memory_checker.go` and  `memory.go` should be merged?

Comment (amyangfei) [reply pkg/util/memory.go:1]: merge them into `memory.go`.

**Comment by amyangfei** [comment]
/run-dm-integration-test
/run-integration-test /tidb=6.5.0 /tikv=6.5.0 /pd=6.5.0 /tiflash=6.5.0
/run-kafka-integration-test /tidb=6.5.0 /tikv=6.5.0 /pd=6.5.0 /tiflash=6.5.0

**Comment by amyangfei** [comment]
/run-integration-test /tidb=v6.5.0 /tikv=v6.5.0 /pd=v6.5.0 /tiflash=v6.5.0
/run-kafka-integration-test /tidb=v6.5.0 /tikv=v6.5.0 /pd=v6.5.0 /tiflash=v6.5.0

**Comment by amyangfei** [comment]
/run-dm-integration-test
/run-integration-test /tiflash=release-6.6 /tikv=release-6.6
/run-kafka-integration-test /tiflash=release-6.6 /tikv=release-6.6

**Comment by amyangfei** [comment]
/run-integration-test /tiflash=release-6.6 /tikv=release-6.6
/run-kafka-integration-test /tiflash=release-6.6 /tikv=release-6.6

**Comment by amyangfei** [comment]
/run-engine-integration-test
/run-kafka-integration-test /tiflash=release-6.6 /tikv=release-6.6

**Pull request merged**